File: lua/hello, world.lua
print("Hello, world")



io.write("\n\nhow many iterations of fibonacci to run? ")
count = tonumber (io.read())
if not count then
    print("invalid input, using default (20)")
    count = 20
end
print()
a, b, c = 0, 1, 1

for i = 0, count do
    print(c)
    c = a+b
    a, b = b, c
end



io.write("\n\nhow many thousands of iterations of leibniz's pi approximation algorithm to run? ")
count = tonumber (io.read())
if not count then
    print("invalid input, using default (20)")
    count = 20
end
print()

a = 1
for i = 3, 4000*count, 4 do
    a = a - 1/i + 1/(i+2)
    if i%4000 == 3 then
        print((a + -1/(i+4))* 4)
    end
end



io.write("\n\nhow many numbers to find the prime factorization of? ")
count = tonumber (io.read())
if not count then
    print("invalid input, using default (10000)")
    count = 10000
end
print()

function findPrimeFactors(num)
    factors = {}
    while num%2 == 0 do
        num = num/2
        factors[#factors] = 2
        Output = Output .. 2 .. ", "
    end
    i = 3
    while i <= num do
        if num % i == 0 then
            num = num/i
            factors[#factors] = i
            Output = Output .. i .. ", "
            i = 3
        else
            i = i + 2
            if i > num/i then
                Output = Output .. num .. ", "
                factors[#factors] = num
                break
            end
        end
    end
    return(factors)
end

a = {}
Output = ""
for i = 0, count do 
    a[i] = math.floor(math.random()*1000000000)
end
initTime = os.time()

for i = 0, #a do
    Output = Output .. tostring(a[i]) .. "\n"
    findPrimeFactors(a[i])
    Output = Output:sub(1, -3) .. "\n\n"
    if i%1000 == 0 then
        print("\r" .. tostring(math.floor(i/#a*10000+0.5)/100) .. "% complete    ")
    end
end

runTime = "got prime factorization of " .. tostring(count) .. " numbers from 1-10^9 in " .. tostring(os.difftime (os.time(), initTime)) .. " seconds"
print("done!")
print(Output)
print(runTime)
io.write("Finished. Press enter to exit.")
count = io.read()

os.exit()

File: C/hello, world.c
#include <stdio.h>
#include <time.h>
#include <math.h>

int count = 0;
int magicPrimeFactorizationDivider = 100000;//handles scale for numbers

void fibonacci() {
    printf("\n\nhow many iterations of fibonacci to run? ");
    scanf("%d", &count);

    if (!count || count < 1) {
        printf("\ninvalid input, using default (20)");
    }

    int a = 0;
    int b, c = 1;

    for (int i = 0; i < count; i++) {
        printf("\n%d", c);
        c = a + b;
        a = b;
        b = c;
    }
}

void leibnizPi() {
    printf("\n\n\nhow many thousands of iterations of leibniz's pi approximation algorithm to run? ");
    scanf("%d", &count);
    if (!count) {
        printf("\ninvalid input, using default (20)");
        count = 20;
    }

    double num = 1;
    double thisIteration;
    for (int i = 3; i < 25000 * count; i += 4) {
        num = num - 1.0 / i + 1.0 / (i + 2);
        if (i % 25000 == 3) {
            thisIteration = (num + -1 / (i + 4)) * 4;
            printf("\n%lf", thisIteration);
        }
    }
    thisIteration = (num) * 4;
    printf("\n%lf", thisIteration);
}

void findPrimeFactors(int product) {
    int factorsLength = 0;
    const int originalProduct = product;

    while (product % 2 == 0) {
        product = product / 2;
        //int small = pow(2, factorsLength+1);
        //int large = (int) originalProduct/small;
        //printf("(%d, %d)", small, large);
        printf("%d ", 2);
        factorsLength++;
    }

    int i = 3;
    while (i <= product) {
        if (product % i == 0) {
            product = product / i;
            //printf("(%d, %d) ", i, originalProduct/i);
            printf("%d ", i);
            i = 3;
            factorsLength++;
        }
        else {
            i = i + 2;
            if (i > product / i) {
                break;
            }
        }
    }
    printf("%d ", product);
}

void primeFactorization() {
    printf("\n\nhow many numbers to find the prime factorization of? ");
    scanf("%d", &count);
    int completeness = 0;

    for (int i = 0; i < count; i++) {
        if (i % 100 == 0) {
            completeness = floor(i / count * 10000 + 0.5) / 100;
        }
        int product = 2 + rand() / magicPrimeFactorizationDivider;

        printf("\n%d%% product: %d, factors: ", completeness, product);
        findPrimeFactors(product);
    }
}

int main(void) {
    srand(time(NULL));

    printf("Hello, world\n");

    fibonacci();
    leibnizPi();
    primeFactorization();
}
